benches/serialization.rs

```diff
@@ -149,3 +173,30 @@ fn do_deserialize_bench(b: &mut Bencher, low: u64, high: u64, digits: u8, fracti
         let _: Histogram<u64> = d.deserialize(&mut cursor).unwrap();
     });
 }
+
+// Maybe someday there will be an obvious right answer for what serialization should look like, at
+// least to the user, but for now we'll only take an easily reversible step towards that. There are
+// still several ways the serializer interfaces could change to achieve better performance, so
+// committing to anything right now would be premature.
+trait TestOnlyHypotheticalSerializerInterface {
```

Comment:
Yeah, I would like that as well but I couldn't think of a way to do it without making it `pub`. Thoughts?